sdlib/d/gc/tcache.d

```diff
@@ -151,34 +163,33 @@ public:
 	 * 
 	 * See also: https://dlang.org/spec/arrays.html#capacity-reserve
 	 */
-	bool getAppendablePageDescriptor(const void[] slice,
-	                                 ref PageDescriptor pd) {
+	bool getAppendablePageDescriptor(const void[] slice, ref PageDescriptor pd,
```

Comment:
None of the callsites use the page descriptor...

Reply:
Wait, setUsedCapacity does. It still seems that's a bit off.

Reply:
What I'm saying is that there are a lot of function in there that only use their argument is certain branch, caller of these function that only use what the return in certain case, ignoring it in most, that there are many repeated checks for the same conditions, and that all of this is indicative of that the problem is not sliced int he right way.